test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,318 @@
+                                await ModuleUtil.ReportTestResultAsync(
+                                    reportClient,
+                                    Logger,
+                                    dmTestResult,
+                                    cts.Token).ConfigureAwait(false);
+                            };
+                        directMethodTask = directMethod();
+                    }
+                    else
+                    {
+                        directMethodTask = Task.CompletedTask;
+                    }
+
+                    // Wait for the two task to be done before do a restart
+                    await Task.WhenAll(new[] { sendMessageTask, directMethodTask });
+
+                    // Wait until the specified restart period to do another restart
+                    await Task.Delay((int)(eachTestExpirationTime - DateTime.UtcNow).TotalMilliseconds, cts.Token);
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Exception caught: {e}");
+                throw;
+            }
+            finally
+            {
+                iotHubServiceClient?.Dispose();
+                dmModuleClient?.Dispose();
+                msgModuleClient?.Dispose();
+            }
+
+            completed.Set();
+            handler.ForEach(h => GC.KeepAlive(h));
+            Logger.LogInformation("EdgeHubRestartTester Main() finished.");
+            return 0;
+        }
+
+        static TestResultBase CreateTestResult(
+            TestOperationResultType testOperationResultType,
+            DateTime edgeHubRestartedTime,
+            HttpStatusCode edgeHubRestartStatusCode,
+            DateTime completedTime,
+            HttpStatusCode completedStatus,
+            Guid batchId,
+            uint restartSequenceNumber,
+            long sequenceNumber)
+        {
+            switch (testOperationResultType)
+            {
+                case TestOperationResultType.Messages:
+                    return new EdgeHubRestartMessageResult(
+                        Settings.Current.ModuleId + testOperationResultType.ToString(),
+                        DateTime.UtcNow,
+                        Settings.Current.TrackingId,
+                        batchId.ToString(),
+                        sequenceNumber.ToString(),
+                        edgeHubRestartedTime,
+                        edgeHubRestartStatusCode,
+                        completedTime,
+                        completedStatus,
+                        restartSequenceNumber);
+
+                case TestOperationResultType.DirectMethod:
+                    return new EdgeHubRestartDirectMethodResult(
+                        Settings.Current.ModuleId + testOperationResultType.ToString(),
+                        DateTime.UtcNow,
+                        Settings.Current.TrackingId,
+                        batchId,
```

Comment:
Please align the usage of batchId here and above; do you need ToString().

Reply:
The underlying `BatchId` field for `MessageTestResult` and `DirectMethodTestResult` have different type which reflects here.